src/Avalonia.Base/Input/InputElement.cs

```diff
@@ -442,6 +443,23 @@ private set
             {
                 SetAndRaise(IsEffectivelyEnabledProperty, ref _isEffectivelyEnabled, value);
                 PseudoClasses.Set(":disabled", !value);
+
+                if (!IsEffectivelyEnabled)
+                {
+                    if (FocusManager.Instance?.Current == this)
+                    {
+                        _restoreFocus = true;
+                        FocusManager.Instance?.Focus(null);
+                    }
+                    else
+                    {
+                        _restoreFocus = false;
+                    }
+                }
+                else if (IsEffectivelyEnabled && _restoreFocus)
+                {
+                    FocusManager.Instance?.Focus(this);
```

Comment:
what should happen in this scenario:

1. control is disabled, looses focus.
2. another control gets focused
3. the original control is enabled again...

At step 3 will the original control steal the focus?

Reply:
Should be fixed now.